src/components/UploadButton/UploadButton.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import './UploadButton.css';
+
+interface PropTypes {
+  // Button contents
+  label?: string;
+  // This type comes from a built-in React method.
+  // Essentially, its typed to the "setState" function.
+  setFile: React.Dispatch<React.SetStateAction<string | null>>;
```

Comment:
Since you pass a string here you could call this `setFileContent` - or maybe you keep it a `setFile` and you pass the File object.

Reply:
Keeping `setFile` makes most sense, and I've changed it so state expects a File.
See commit: https://github.com/AsyncRacing/AsyncRacing/pull/5/commits/9cf04aea32863163883988680bed505422e637b8